ddtrace/contrib/internal/selenium/patch.py

```diff
@@ -0,0 +1,185 @@
+        try:
+            self._handle_return()
+        except Exception:  # noqa: E722
+            log.debug("Error handling instrumentation return", exc_info=True)
+
+        return value
+
+
+class SeleniumGetWrappingContext(SeleniumWrappingContextBase):
+    def _handle_return(self) -> None:
+        root_span = ddtrace.tracer.current_root_span()
+        test_trace_id = root_span.trace_id
+
+        if root_span is None or root_span.get_tag("type") != "test":
+            return
+
+        webdriver_instance = self._get_webdriver_instance()
+
+        if webdriver_instance is None:
+            return
+
+        # The trace IDs for Test Visibility data using the CIVisibility protocol are 64-bit
+        # TODO[ci_visibility]: properly identify whether to use 64 or 128 bit trace_ids
+        trace_id_64bit = test_trace_id % 2**64
+
+        webdriver_instance.add_cookie({"name": "datadog-ci-visibility-test-execution-id", "value": str(trace_id_64bit)})
+
+        root_span.set_tag("test.is_browser", "true")
+        root_span.set_tag("test.browser.driver", "selenium")
+        root_span.set_tag("test.browser.driver_version", get_version())
+
+        # Submit empty values for browser names or version if multiple are found
+        browser_name = webdriver_instance.capabilities.get("browserName")
+        browser_version = webdriver_instance.capabilities.get("browserVersion")
+
+        existing_browser_name = root_span.get_tag("test.browser.name")
+        if existing_browser_name is None:
+            root_span.set_tag("test.browser.name", browser_name)
+        elif existing_browser_name not in ["", browser_name]:
+            root_span.set_tag("test.browser.name", "")
+
+        existing_browser_version = root_span.get_tag("test.browser.version")
+        if existing_browser_version is None:
+            root_span.set_tag("test.browser.version", browser_version)
+        elif existing_browser_version not in ["", browser_version]:
+            root_span.set_tag("test.browser.version", "")
+
+
+class SeleniumQuitWrappingContext(SeleniumWrappingContextBase):
+    def _handle_enter(self) -> None:
+        root_span = ddtrace.tracer.current_root_span()
+
+        if root_span is None or root_span.get_tag("type") != "test":
+            return
+
+        webdriver_instance = self._get_webdriver_instance()
+
+        if webdriver_instance is None:
+            return
+
+        is_rum_active = webdriver_instance.execute_script(_RUM_STOP_SESSION_SCRIPT)
+        time.sleep(config.selenium.flush_sleep_ms / 1000)
+
+        if is_rum_active:
+            root_span.set_tag("test.is_rum_active", "true")
+
+        webdriver_instance.delete_cookie("datadog-ci-visibility-test-execution-id")
+
+
+def get_version() -> str:
+    import selenium
+
+    try:
+        return selenium.__version__
+    except AttributeError:
+        log.debug("Could not get Selenium version")
+        return ""
+
+
+def patch() -> None:
```

Comment:
Could we add a test that calls this patch method (I see you have a test that calls unpatch)

Reply:
None of the other tests that validate the functionality work if `patch()` doesn't work. Adding a test just for `patch()` doesn't seem valuable to me.

In fact, the `unpatch()` test doesn't test that the methods are patched, it tests that, when the functions are called after unpatching, we don't continue to add tags.